rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -1013,16 +1009,15 @@ def training_states_labels_and_entities(
             Trackers as states, labels, and entity data.
         """
         example_states = []
-        example_labels = []
         example_entities = []
 
         # Store of example hashes for removing duplicate training examples.
         hashed_examples = set()
         # Mapping of example state hash to list of positive labels associated with
-        # the state. Note that each individual 'label' instance is a list of ints.
-        state_hash_to_label_list_instances: defaultdict[
-            int, List[List[int]]
-        ] = defaultdict(list)
+        # the state. Note that each individual 'label' instance is a list of strings.
+        state_hash_to_label_set: defaultdict[int, Set[Text]] = defaultdict(set)
+
+        seen_states = set()
```

Comment:
```suggestion
        # Store of example hashes (of both states and labels) for removing 
        # duplicate training examples.
        hashed_examples = set()
        # Mapping of example state hash to set of positive labels associated with
        # the state. Note that each individual 'label' instance is a set of strings.
        state_hash_to_label_set: defaultdict[int, Set[Text]] = defaultdict(set)
        # Store of example state hashes so only unique states are kept when 
        # removing duplicates.
        seen_states = set()
```

Added a comment for `seen_states` and modified comments of `state_hash_to_label_set`.

Reply:
Omitting ```Note that each individual 'label' instance is a set of strings.``` since that just mentions the data type which is already known by the type annotation.